### PR TITLE
Update dependencies

### LINKED ISSUE #215: Upgrade dependencies
### Context

Upgrade all dependencies, and fix interfaces problems.

### Example
 
Issue #209  is a good example of a small problem caused by an outdated dependency. 
And the usage of old tools like `webpack` and `babel` keep us tied to a bunch of small problems which could break everything sometimes keeping all updated we also avoid a lot of security threats 
and improve some tools performance like (Storybook)[https://github.com/storybooks/storybook/].

### Description

This issue aims to upgrade all `dependencies` and all `devDependencies` of the project. The upgrade process may create some errors due to the dependencies interfaces, these errors should be solved in the upgrade process.
The major upgrades will be:
`babel` from version 6.26.0 to version 7.2.2
`jest` from version 22.1.4  to version  23.6.0 
`react` from version 16.2.0 to version 16.7.0
`storybook` from version 3.3.11  to version 4.1.7 ( this dependency is splited in 7 parts)
`webpack` from version 3.10.0 to version 4.28
 
### Requirements

- [x] Upgrade all outdated dependencies.
- [x] Refactor new dependencies interfaces usages.

### PR DESCRIPTION
**Pull request by Deivis**
## Context
This PR aims to upgrade a lot of project dependencies, must of these dependencies are will suffer major updates which could break some interfaces.

## Checklist
- [x] Update major dependencies
- [x] Refactor dependencies usage

## Linked Issues
- [x] resolves #215 

## How to test this
1 - Download the PR branch
2 - Go to the project folder, delete `yarn.lock` and `node_modules`(just in case 😄 )
3 - Download the `former-kit-skin-pagarme` branch https://github.com/pagarme/former-kit-skin-pagarme/pull/156 and build it using:
```
yarn build
```
4 - Create a link in the `former-kit-skin-pagarme` using the following command
```
yarn link
```
5 - Link the `former-kit-skin-pagarme` in the `former-kit` project
```
yarn link former-kit-skin-pagarme
```
6 - Install all dependencies using the following command
```
yarn
```
7 - Start the _Storybook_ process using:
```
yarn storybook
```

The _Storybook_ should run without warnings.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 0f79d4c46bb2ea50e2f151e1ac91b0627f1fc784
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

 * Commit: 24bb7129cd1e3bb9cf6d29a579b603ca60720699
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: eeed7a90a6e02f4d05d92a36867f84ff2fef879a
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: eeed7a90a6e02f4d05d92a36867f84ff2fef879a
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: df6bb157e939e0c0b698e31df7c3fda9c1258d06
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: df6bb157e939e0c0b698e31df7c3fda9c1258d06
  * lways use the standard `scope: action description`, sempre em inglês  
    - The **scope** shouldn't have spaces. If the word is composite, use dashes.  
    - The **action** should be **add**, **remove**, **refactor**, **move**,  **bump**, **update** or **release**  
    - The **description** should contain a quick description of the change

The complete commit style guide is available at https://github.com/pagarme/git-style-guide

---
This message was auto-generated by https://gitcop.com

File: .babelrc [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "presets": [
-    "react-app"
-  ],
-  "plugins": [
-    "babel-plugin-transform-es2015-modules-commonjs"
+    ["@babel/preset-env", {
+      "loose": true,
```
Comment (vagnervst) [inline .babelrc]: Should we specify this option here, considering we didn't do it in the past configuration of `.babelrc` (defaulting to `false`)? 🤔

File: src/Modal/Modal.js [hunk 2]
```diff
@@ -65,7 +69,15 @@ Modal.propTypes = {
 
 Modal.defaultProps = {
   onRequestClose: null,
-  theme: {},
+  theme: {
```
[the remark this answers is not in the document]
Comment (Deivis) [reply src/Modal/Modal.js]: Boa pergunta por que não definimos isso antes, mas não acho que isso seja um problema agora 😄

**Pull request merged**